backend/typescript/rest/authRoutes.ts

```diff
@@ -142,4 +149,14 @@ authRouter.post(
   },
 );
 
+/* verifies a user by the given uid in the url */
+authRouter.post("/verifyEmail/:uid", async (req, res) => {
```

Comment:
Im assuming these routes have some sort of user authentication right (to make sure random people cant make requests)?

Reply:
we can't do authentication check because those end points are called when the user is not authenticated (they are used when the user is setting up their account for the first time). The logic we used was that it's hard to get access to the uid since it's a bunch of random letters and numbers. But you're right that it's technically  a security hazard, we just couldn't think of a way to also protect it.

Do you have any suggestions on protecting the route without needing to be logged in?

Reply:
This is a good point, should we add the `isAuthenticatedById` middleware here?

Reply:
One way to do this would have the link in the email contain some sort of hash that the backend checks. This way a random user cant just verify themselves. But given this a security issue it might be worth just not doing for now if there is no other easier way

Reply:
@susheelk we can't add the isAuthenticatedById middleware here because the route needs to be accessible when you are not logged in; hence, the user won't be authenticated when they are verifying their account (since the access token won't be set). 

Currently, there is only 1 way this functionality is protected: only the uid is in the route- this is a hard to guess random combination of letters and numbers. Along with the general encryption of the HTTP requests I guess.

Reply:
@nimwijetunga I like the hash idea. Currently, the "access code" is just the old password that the user has set prior to actually verifying and setting up their password. We can pass this existing password (aka: "access code") to the verifyEmail function in the authService which will first check the access code and then verify the user. 

This way, to verify a user, you need both their UID and their password. What do you think about that?

Reply:
Oops didn't read your reply @dhmi19. I like the hash idea as well (maybe we could add some sort of nonce/salt as well) but that should probably be a separate ticket

Reply:
> @nimwijetunga I like the hash idea. Currently, the "access code" is just the old password that the user has set prior to actually verifying and setting up their password. We can pass this existing password (aka: "access code") to the verifyEmail function in the authService which will first check the access code and then verify the user.
> 
> This way, to verify a user, you need both their UID and their password. What do you think about that?

So would this existing password be included in the verify email link? If so it would be hashed right?

Reply:
Currently, the verifyEmail function is called by a POST request to '/auth/verifyEmail/${uid}' route. We can change this so that it is a POST request to '/auth/verifyEmail' route and have the uid and temp password (aka: access code) in the body. As I understand it, the HTTP request encrypts everything so the info will be protected. We can hash it if we don't want even us to see the temp password, however, seeing that the user will change it almost instantly after the request is made, idk if that's necessary? 

Let me know what you think :)